R/05_analysis_1.qmd

```diff
@@ -0,0 +1,82 @@
+
+```{r}
+    #| message: false
+    library(dplyr)
+    library("tidyverse")
+    rm(list = ls())
+
+
+diabetes_data <- read_tsv(file = "../data/04_diabetes_data_augment.tsv")
+```
+
+```{r}
+plot1 <- diabetes_data |>
+                  mutate(income = factor(income, levels = c('[0,5000)','[5000,10000)','[10000,15000)','[15000,20000)','[20000,25000)',
+                          '[25000,35000)','[35000,45000)','[45000,55000)','[55000,65000)','[65000,75000)','[75000,100000)'))) |>
+                  filter(!is.na(income)) |>
+                  group_by(income) |>
+                  summarize(percentage = sum(diagnosis == 1)*100 / n())
+
+plot1 <- ggplot(plot1, aes(x = income, y = percentage, alpha = percentage)) +
+  geom_bar(stat="identity", color="black", fill='wheat', show.legend=FALSE) +
+  scale_alpha_continuous(range = c(0.2, 0.8)) +
+  theme(text = element_text(size=10), axis.text.x = element_text(angle = 15, vjust = 1, hjust=1)) +
+  coord_cartesian(ylim = c(0, 100)) +
+  labs(
+    title = "Diabete-diagnosed people vs Family income classes",
+    x = "Income classes",
+    y = "% of diabete-diagnosed",
+    caption = "To be read: within the class of income X, Y% of individuals got diagnosed with diabete."
+  )
+
+
+plot1
+```
+
+
+```{r}
+plot2 <- diabetes_data |>
+                  mutate(income = factor(income, levels = c('[0,5000)','[5000,10000)','[10000,15000)','[15000,20000)','[20000,25000)',
+                        '[25000,35000)','[35000,45000)','[45000,55000)','[55000,65000)','[65000,75000)','[75000,100000)'))) |>
+                  filter(!is.na(income), diagnosis==1) |>
+                  group_by(income) |>
+                  summarize(percentage = sum(medication == 1)*100 / n())
+
+plot2 <- ggplot(plot2, aes(x = income, y = percentage, alpha = percentage)) +
+  geom_bar(stat="identity", color="black", fill='wheat', show.legend=FALSE) +
+  scale_alpha_continuous(range = c(0.2, 0.8)) +
+  theme(text = element_text(size=10), axis.text.x = element_text(angle = 15, vjust = 1, hjust=1)) +
+  coord_cartesian(ylim = c(0, 100)) +
+  labs(
+    title = "Medicated people vs Family income classes",
+    x = "Income classes",
+    y = "% of medicated people",
+    caption = "To be read: within the class of income X, Y% of (diabete-diagnosed) individuals got medicated."
+  )
+
+plot2
+```
+
+```{r}
+plot3 <- diabetes_data |>
+                  arrange(age) |>
+                  group_by(age) |>
+                  summarize(percentage = sum(diagnosis == 1)*100 / n())
+
+plot3 <- ggplot(plot3, aes(x = age, y = percentage, alpha = percentage)) +
+  geom_bar(stat="identity", color="black", fill="wheat", show.legend=FALSE) +
+  scale_alpha_continuous(range = c(0.2, 0.8)) +
```

Comment:
Why also scale alpha, since it is already chosen?

Reply:
I wanted alpha to vary with the level of percentage :) but not from 0 to 1 because otherwise some bars would have been blank: therefore I limited the opacity within 0.2 and 0.8. I'm editing all the rest btw